### ISSUE TITLE
ctc_loss.cpp:509:10: fatal error: 'ctc.h' file not found

### ISSUE THREAD

**Issue by eravallirao** [Bartzi/stn-ocr#28]
![Screenshot 2019-06-17 at 4 44 56 PM](https://user-images.githubusercontent.com/49031948/59600617-57872880-911f-11e9-8538-988770b61b18.png)
I am getting the above warning while running "make"

After that while running this command python3 setup.py build_ext --inplace I am getting the below error.
![Screenshot 2019-06-17 at 4 46 49 PM](https://user-images.githubusercontent.com/49031948/59600715-9cab5a80-911f-11e9-8887-6e7539685d42.png)
Can you please help me with this

**Comment by Bartzi** [comment]
did you run `make install` after building `warp_ctc`? Did you make sure that Cython is able to find the header files of `warp_ctc`.

**Comment by eravallirao** [comment]
![Screenshot 2019-06-17 at 4 53 40 PM](https://user-images.githubusercontent.com/49031948/59601025-7df99380-9120-11e9-8223-fa6990b546b7.png)
I am getting this while running make install

**Comment by eravallirao** [comment]
I tried giving "sudo" 
![Screenshot 2019-06-17 at 4 55 37 PM](https://user-images.githubusercontent.com/49031948/59601132-c5801f80-9120-11e9-97da-14a517347c17.png)

**Comment by eravallirao** [comment]
**Did you make sure that Cython is able to find the header files of warp_ctc?**
How can we check this? I am little bit new to this, please don't mind if I am asking for basics.

**Comment by Bartzi** [comment]
if your `CMAKE_INSTALL_PREFIX` is set to `/usr/local`. Cython should be able to find your files, once you did run `make install` successfully.

**Comment by eravallirao** [comment]
I followed the path you asked to, still I am getting the error of ctc.h not found

**Comment by eravallirao** [comment]
Do you have any docker setup available

**Comment by Bartzi** [comment]
did you check that `ctc.h` is in `/usr/local/include`?
Unfortunately, I can not provide a docker image at the moment.

**Comment by eravallirao** [comment]
Sorry for late reply.
Yeah I did check, It was not there. I copied the ctc.h file and copied into the path you specified and ran it. I am getting other error
![Screenshot 2019-06-20 at 11 57 49 AM](https://user-images.githubusercontent.com/49031948/59825864-c4363900-9352-11e9-9f94-1a8117e89868.png)

**Comment by Bartzi** [comment]
It seems that the linker can not find the library. Where is your compiled library stored on your pc? Is that path in your environment variable `LD_LIBRARY_PATH`?

**Comment by eravallirao** [comment]
Sorry can you please tell me about the compiled library? FYI I don't have CUDA. I am running it on MAC

**Comment by eravallirao** [comment]
sukshis-MacBook-Air-2:ctc Puru$ echo $LD_LIBRARY_PATH
/Users/sukshi/torch/install/lib:

**Comment by Bartzi** [comment]
the compiler creates a `.so` file once you are done with compiling the `warpctc` library. That is the file I'm talking about. If you place this file in `/Users/sukshi/torch/install/lib` the compiler, compiling the python extension, might find it.

**Comment by eravallirao** [comment]
![Screenshot 2019-06-20 at 2 00 37 PM](https://user-images.githubusercontent.com/49031948/59833506-cfde2b80-9363-11e9-9b54-d41202ac34ed.png)
Are you talking about the same .so file which is in that directory?

**Comment by Bartzi** [comment]
No, I'm talking about a file that sshould be named `libwarpctc.so` or something like that. I see that you have a `libwarpctc.dylib` file. This might be a mac equivalent of a `.so` file but I don't know anything about macs and I can not help you with that. In the Linux world you'll need a `.so` file for the linker but how that stuff works on mac (if it is different at all) I really don't know.
